app/src/main/java/com/mailerdaemon/app/LoginActivity.kt

```diff
@@ -0,0 +1,195 @@
+package com.mailerdaemon.app
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.content.Intent
+import android.content.SharedPreferences
+import android.os.Build
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.facebook.AccessToken
+import com.facebook.CallbackManager
+import com.facebook.FacebookCallback
+import com.facebook.FacebookException
+import com.facebook.login.LoginManager
+import com.facebook.login.LoginResult
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.tasks.Task
+import com.google.firebase.auth.*
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.android.synthetic.main.activity_login.*
+import java.util.*
+
+@Suppress("NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
+class LoginActivity: AppCompatActivity() {
+
+    private val email = "email"
+    private val tag = "LoginActivity"
+    private val rcsignin = 234
+    private lateinit var mAuth: FirebaseAuth
+    private lateinit var callbackManager: CallbackManager
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setTheme(R.style.AppTheme_NoActionBar_NoStatusColor)
+        val mAuth = FirebaseAuth.getInstance()
+        if (getSharedPreferences("MAIN", Context.MODE_PRIVATE).getBoolean("intro", true)) {
+            startActivity(Intent(this, IntroActivity::class.java))
+            finish()
+        } else {
+            val currentUser = mAuth.currentUser
+            currentUser?.let { startMain() }
+        }
+        setContentView(R.layout.activity_login)
+        progress_bar.visibility = View.GONE
+        forgot_password.setOnClickListener { startActivity(Intent(this, ForgotPassActivity::class.java)) }
+        login.setOnClickListener {
+            val email = Objects.requireNonNull(login_email.text).toString().trim { it <= ' ' }
+            val password = Objects.requireNonNull(login_password.text).toString()
```

Comment:
Replace `Objects.requireNonNull' with suitable expression. Same for all other places.